### PR TITLE
add ascatngs as submodule and fixing tests

### PR DESCRIPTION
**Pull request by hassanfa**
#### Added:

- ascatngs as submodule

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
@khurrammaqbool @ashwini06 this PR turned out to be different than ascatNGS. While fixing ascat's actions, i noticed pytest actions is not working because of a typo and failed silently.

I had to fix bunch of other test for recent changes.

File: .github/workflows/docker_build_push_release_submodules.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        id: get_branch_name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF##*/})"
+      - name: Login to DockerHub
+        id: docker_login
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build only
```
Comment (ashwini06) [inline .github/workflows/docker_build_push_release_submodules.yml:29]: Is it Build and push ??

File: .github/workflows/docker_build_push_submodules.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        id: get_branch_name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF##*/})"
+      - name: Login to DockerHub
+        id: docker_login
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build only
```
Comment (ashwini06) [inline .github/workflows/docker_build_push_submodules.yml:30]: Same here.. Is it Build and push ? or else in line 35: `push: false`?

File: BALSAMIC/snakemake_rules/variant_calling/cnvkit_single.rule [hunk 3]
```diff
@@ -41,6 +40,7 @@ rule cnvkit_single:
 source activate {params.conda};
 mkdir -p {params.tmpdir};
 export TMPDIR={params.tmpdir};
+export PURECN='/opt/conda/envs/varcall_cnvkit/lib/R/library/PureCN/extdata'
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/cnvkit_single.rule:43]:
```suggestion
export PURECN='/opt/conda/envs/varcall_cnvkit/lib/R/library/PureCN/extdata/PureCN.R'
```

**Review by ashwini06 — COMMENTED** [review]
Thanks for fixing cnvkit changes as well :) Few comments..

**Comment by khurrammaqbool** [comment]
We should give it a try to place submodules in a folder.

**Pull request merged**